src/libgit2/sparse.c

```diff
@@ -159,6 +159,45 @@ static int parse_sparse_file(
 		}
 	}
 
+	// Check for positive patterns that have their parent excluded.
+	// Ex: "A/B/C" without "A/B" is considered invalid.
+	git_vector_foreach(&attrs->rules, j, match) {
```

Comment:
Would it be much more challenging to have it match the git behavior?

E.g., if there are positive patterns that have their parent excluded to add them when parsing the file?

Reply:
The git behavior is inconsistent because `/A/B/` automatically includes `A/`, but `/A/B/C/` doesn't include `A/` or `A/B/`. To me it seems like a bug, as if in that second case, git is checking for "A" and thinking "oh, it's a top level file, I'll check deeper! Then goes into "A/B" and is like "nope, it doesn't match `/A/B/C/`, skip". 

If it was consistent I wouldn't have a problem implementing it, but we have the choice to:
- copy the behavior (breaking mental model of cone and making the logic more complex)
- change the behavior to include deeper paths (inconsistent when used with git), 
- error out (no downsides!, especially if we add a command line interface to manage the patterns)

Reply:
I'm fine with erroring out for now.